openfl/transport/grpc/aggregator_server.py

```diff
@@ -206,9 +209,102 @@ def SendLocalTaskResults(self, request, context):  # NOQA:N802
+        elif request.model_type == aggregator_pb2.GetTrainedModelRequest.LAST_MODEL:
+            model_type = 'last'
+        else:
+            logger.error('Incorrect model type')
+            return aggregator_pb2.TrainedModelResponse()
+
+        trained_model_dict = None
+        if self.aggregator.last_tensor_dict is None:
+            logger.error('Aggregator have no aggregated model to return')
+        elif model_type == 'best':
+            trained_model_dict = self.aggregator.best_tensor_dict
+        elif model_type == 'last':
+            trained_model_dict = self.aggregator.last_tensor_dict
+
+        if trained_model_dict is None:
+            return aggregator_pb2.TrainedModelResponse()
+
+        model_proto = utils.construct_model_proto(
+            trained_model_dict, 0, NoCompressionPipeline()
+        )
+
+        return aggregator_pb2.TrainedModelResponse(model_proto=model_proto)
+
+    def GetExperimentDescription(self, request, context):  # NOQA:N802
+        """Get an experiment description."""
+        experiment = self.aggregator.get_experiment_description()
+        models_statuses = [
+            base_pb2.DownloadStatus(
+                name=ms['name'],
+                status=ms['status']
+            )
+            for ms in experiment['download_statuses']['models']
+        ]
+        logs_statuses = [
+            base_pb2.DownloadStatus(
+                name=ls['name'],
+                status=ls['status']
+            )
+            for ls in experiment['download_statuses']['logs']
+        ]
+        download_statuses = base_pb2.DownloadStatuses(
+            models=models_statuses,
+            logs=logs_statuses,
+        )
+        collaborators = [
+            base_pb2.CollaboratorDescription(
+                name=col['name'],
+                status=col['status'],
+                progress=col['progress'],
+                round=col['round'],
+                current_task=col['current_task'],
+                next_task=col['next_task']
+            )
+            for col in experiment['collaborators']
+        ]
+        tasks = [
+            base_pb2.TaskDescription(
+                name=task['name'],
+                description=task['description']
+            )
+            for task in experiment['tasks']
+        ]
+
+        return aggregator_pb2.GetExperimentDescriptionResponse(
+            experiment=base_pb2.ExperimentDescription(
+                progress=experiment['progress'],
+                current_round=experiment['current_round'],
+                total_rounds=experiment['total_rounds'],
+                download_statuses=download_statuses,
+                collaborators=collaborators,
+                tasks=tasks,
+            ),
+        )
+
     def get_server(self):
         """Return gRPC server."""
-        self.server = server(ThreadPoolExecutor(max_workers=cpu_count()),
+        self.server = server(ThreadPoolExecutor(max_workers=cpu_count() + 1),
```

Comment:
To avoid case when we have only CPU and only thread in the thread pool. Moreover, it looks strange, that the number of thread depends from CPU count.